src/Bicep.Core/Diagnostics/DiagnosticBuilder.cs

```diff
@@ -400,6 +401,21 @@ public ErrorDiagnostic ArgumentCountMismatch(int argumentCount, int mininumArgum
                 TextSpan,
                 "BCP077",
                 $"The property '{badProperty}' on type '{type}' is write-only. Write-only properties cannot be accessed.");
+
+            public ErrorDiagnostic MissingDiscriminator(string discriminatorKey, IEnumerable<string> keys) => new ErrorDiagnostic(
+                TextSpan,
+                "BCP078",
+                $"Property '{discriminatorKey}' must be supplied. Valid key types are: {string.Join(", ", keys.OrderBy(x => x))}.");
```

Comment:
>key [](start = 72, length = 3)

What's a "key"? We already have an error for a required property that could be updated to also include type of the property. Do we need another error like this just for discriminators?

Reply:
The "key" here is the discriminator. Agree that's not very clear from this message. Let me take a look and see if I can combine the two error messages.

Reply:
Yup, I know what you meant - was just pretending to be a user unfamiliar with this.

Reply:
Makes sense. I've pushed some updates - hopefully to clear up the messages.